app/models/upload_reference.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class UploadReference < ActiveRecord::Base
+  belongs_to :upload
+  belongs_to :target, polymorphic: true
+end
+
+# == Schema Information
+#
+# Table name: upload_references
+#
+#  id          :bigint           not null, primary key
+#  upload_id   :bigint
+#  target_type :string
+#  target_id   :bigint
```

Comment:
I think these columns should be `NOT NULL` too?

Reply:
Yeah these definitely need to be NOT NULL, all of them

Reply:
I think we should add `timestamps` to all tables we create too, it's a pain when they are not there when it comes to debugging issues.